server/main.py

```diff
@@ -18,4 +185,57 @@ async def create_upload_file(file: UploadFile = File(...)):
         "filename": file.filename,
         "content_type": file.content_type,
         "location": f"../data/syllabus/{file.filename}",
-    }
\ No newline at end of file
+    }
+
+
+# Load documents into Pinecone
+@app.post("/load_documents")
+async def load_documents(load_documents: LoadDocuemnts):
+    crn = None
+    for item in data_map:
+        if item["course_name"] == load_documents.class_name:
+            crn = item["crn"]
+            break
+    syllabus_directory = "../data/syllabus"
+    index_name = f"cs{crn.lower()}-index"
+    pdf_path = None
+    for item in data_map:
+        if item["index"] == index_name:
+            pdf_path = f"{syllabus_directory}/{item['filename']}"
+            break
+    status = await add_pdf_to_pinecone_index(pdf_path, index_name, embeddings)
+    return {"status": status}
+
+
+# Get q&a for a class classname w/ query
+@app.post("/response")
+async def get_response(get_response: GetResponse):
+    index_name = f"cs{get_response.class_name.lower()}-index"
+    docsearch = Pinecone.from_existing_index(index_name, embeddings)
+    docs = docsearch.similarity_search(get_response.query)
+    llm = OpenAI(temperature=0, openai_api_key=OPENAI_API_KEY)
+    chain = load_qa_chain(llm, chain_type="stuff")
+    llm_response = chain.run(input_documents=docs, question=get_response.query)
+    return {"response": llm_response}
+
+
+# List all indexes
+@app.get("/list_indexes")
+async def list_indexes():
+    return pinecone.list_indexes()
+
+
+# Get information about an index
+@app.get("/list_index_info/{index_name}")
+async def list_index_info(index_name: str):
+    return pinecone.describe_index(index_name)
+
+
+# Get all available data
+@app.get("/available_data")
+def available_data():
+    return data_map
+
+# Uvicorn isn't required, but it's a nice way to run a webserver in python
```

Comment:
i don't know how i feel about using uvicorn here since there's no windows support. however, i believe in being able to find an alternate which works just as well. it's not mandatory.